src/sdc11073/xml_types/xml_structure.py

```diff
@@ -805,9 +846,12 @@ def update_xml_value(self, instance: Any, node: etree_.ElementBase):
 
 def _compare_extension(left: etree_.ElementBase, right: etree_.ElementBase) -> bool:
     # xml comparison
-    if left.tag != right.tag:  # compare expanded names
-        return False
-    if dict(left.attrib) != dict(right.attrib):  # unclear how lxml _Attrib compares
+    try:
+        if left.tag != right.tag:  # compare expanded names
+            return False
+        if dict(left.attrib) != dict(right.attrib):  # unclear how lxml _Attrib compares
+            return False
+    except AttributeError:  # right side is not an Element type because expected attributes are missing
```

Comment:
instead of returning false, should return `NotImplemened`. check both instances and if one is not `etree_._Element` then return `NotImplemented`

Reply:
Technically you cant know this, if both are not of type `_Element` thats why you should do an `isinstance` check and return `NotImplemented`. then you dont need the `try-except` block